cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java

```diff
@@ -101,6 +101,57 @@ public void testIsNoOpWhenThrottleIsNull() {
     throttleHelper.clearThrottles(Collections.singletonList(task), Collections.emptyList());
   }
 
+  @Test
+  public void testSetThrottleOnNonExistentTopic() {
+    final long throttleRate = 100L;
+    final int brokerId0 = 0;
+    final int brokerId1 = 1;
+    final int brokerId2 = 2;
+    final int partitionId = 0;
+    // A proposal to move a partition with 2 replicas from broker 0 and 1 to broker 0 and 2
+    ExecutionProposal proposal = new ExecutionProposal(new TopicPartition(TOPIC0, partitionId),
+                                           100,
+                                                       new ReplicaPlacementInfo(brokerId0),
+                                                       Arrays.asList(new ReplicaPlacementInfo(brokerId0), new ReplicaPlacementInfo(brokerId1)),
+                                                       Arrays.asList(new ReplicaPlacementInfo(brokerId0), new ReplicaPlacementInfo(brokerId2)));
+
+    KafkaZkClient mockKafkaZkClient = EasyMock.mock(KafkaZkClient.class);
+    prepareKafkaZkClientMockWithBrokerConfigs(mockKafkaZkClient);
+    // Case 1: a situation where Topic0 does not exist. Hence no property is returned upon read.
+    EasyMock.expect(mockKafkaZkClient.getEntityConfigs(ConfigType.Topic(), TOPIC0)).andReturn(new Properties()).times(2);
+    EasyMock.expect(mockKafkaZkClient.topicExists(TOPIC0)).andReturn(false).times(2);
+    EasyMock.replay(mockKafkaZkClient);
+    ReplicationThrottleHelper throttleHelper = new ReplicationThrottleHelper(mockKafkaZkClient, throttleRate);
+
+    throttleHelper.setThrottles(Collections.singletonList(proposal));
+    EasyMock.verify(mockKafkaZkClient);
+
+    // Case 2: a situation where Topic0 gets deleted after its configs were read. Change configs should not fail.
+    EasyMock.reset(mockKafkaZkClient);
+    prepareKafkaZkClientMockWithBrokerConfigs(mockKafkaZkClient);
+    Properties topicConfigProps = new Properties();
+    String throttledReplicas = brokerId0 + "," + brokerId1;
+    topicConfigProps.put(ReplicationThrottleHelper.LEADER_THROTTLED_REPLICAS, throttledReplicas);
+    topicConfigProps.put(ReplicationThrottleHelper.FOLLOWER_THROTTLED_REPLICAS, throttledReplicas);
+    EasyMock.expect(mockKafkaZkClient.getEntityConfigs(ConfigType.Topic(), TOPIC0)).andReturn(topicConfigProps).times(2);
+    EasyMock.expect(mockKafkaZkClient.topicExists(TOPIC0)).andReturn(false).times(4);
+    EasyMock.replay(mockKafkaZkClient);
+
+    throttleHelper.setThrottles(Collections.singletonList(proposal));
```

Comment:
Should we test the changes in `removeThrottledReplicasFromTopic`, too?

Reply:
Yes, test added